Shoko.Server/API/v3/Models/Shoko/Series.cs

```diff
@@ -397,59 +399,207 @@ private static void AddMovieDBImages(HttpContext ctx, Images images, int animeID
+                var anime = series != null ? series.GetAnime() : RepoFactory.AniDB_Anime.GetByAnimeID(result.AnimeID);
+
+                ID = result.AnimeID;
+                ShokoID = series?.AnimeSeriesID;
+                Type = Series.GetAniDBSeriesType(anime?.AnimeType);
+                Title = series?.GetSeriesName() ?? anime?.PreferredTitle ?? result.MainTitle;
+                Titles = includeTitles ? result.Titles.Select(title => new Title
                     {
+                        Language = title.TitleLanguage,
                         Name = title.Title,
-                        Language = title.Language,
                         Type = title.TitleType,
-                        Default = string.Equals(title.Title, anime.MainTitle),
+                        Default = string.Equals(title.Title, Title),
                         Source = "AniDB",
                     }
-                ).ToList();
+                ).ToList() : null;
+                Description = anime?.Description;
+                Restricted = anime != null ? anime.Restricted == 1 : false;
+                EpisodeCount = anime?.EpisodeCount;
+            }
 
-                if (anime.AirDate != null)
+            public AniDB(SVR_AniDB_Anime_Relation relation, bool includeTitles) : this(relation, null, includeTitles) { }
+
+            public AniDB(SVR_AniDB_Anime_Relation relation, SVR_AnimeSeries series = null, bool includeTitles = true)
+            {
+                if (series == null)
+                    series = RepoFactory.AnimeSeries.GetByAnimeID(relation.RelatedAnimeID);
+                ID = relation.RelatedAnimeID;
+                ShokoID = series?.AnimeSeriesID;
+                if (RepoFactory.AniDB_Anime.TryGetByAnimeID(relation.RelatedAnimeID, out var anime))
                 {
-                    var airdate = anime.AirDate.Value;
-                    if (airdate != DateTime.MinValue)
-                        AirDate = airdate;
+                    Type = GetAniDBSeriesType(anime.AnimeType);
+                    Title = series?.GetSeriesName() ?? anime.PreferredTitle;
+                    Titles = includeTitles ? anime.GetTitles().Select(title => new Title
+                        {
+                            Name = title.Title,
+                            Language = title.Language,
+                            Type = title.TitleType,
+                            Default = string.Equals(title.Title, Title),
+                            Source = "AniDB",
+                        }
+                    ).ToList() : null;
+                    Description = anime.Description;
+                    Restricted = anime.Restricted == 1;
+                    EpisodeCount = anime.EpisodeCountNormal;
                 }
-                if (anime.EndDate != null)
+                else if (AniDB_TitleHelper.Instance.TrySearchAnimeID(relation.RelatedAnimeID, out var result))

```

Comment:
A similar thing as above can be done here, but IDK the exact method naming off the top of my head